### ISSUE TITLE
join lines command should remove the comment characters from the second line

### ISSUE THREAD

**Issue by hferreiro** [zed-industries/zed#4481]
### Summary

`editor::JoinLines` should be comment-aware / language aware any lines after the first should that begin with the language's `line_comments` should be replaced with a space.

### Description

This could be implemented as an optional parameter (e.g. `language_comment_aware`) for the `editor::JoinLines` action.  Not sure whether it should default to true or not (probably?).

Steps to reproduce:

Using the "join lines" command in a Python comment like the following:
```
# This is
# a comment
```
results in:
```
# This is # a comment
```
Instead, it should result in
```
# This is a comment
```

### Environment

Zed: v0.102.2 (stable)
OS: macOS 13.5.2
Memory: 32 GiB
Architecture: aarch64

**Comment by hferreiro** [comment]
Still an issue.

**Comment by miguelraz** [comment]
Sorry all for the noise, the bot got an issue upstream and went haywire: https://github.com/actions/stale/issues/1302.

**Comment by hferreiro** [comment]
Yes it appears to be working. I just found one case where it doesn't:
```
/* multi line
 * comment
 */
```
becomes
```
/* multi line comment /
```

**Comment by ian-h-chamberlain** [comment]
Might this issue be the result of these recent changes? On `0.226.3+preview.188.408db8550ce68f557811a2a2854e6e68590d476b` I'm seeing 

```rs
fn main() {
    let s = todo!();
    // bar
}
```

Moving the cursor to `let s todo!()` and running the join lines command results in 
```rs
fn main() {
    let s = todo!(); bar
}
```

where I would expect the comment to remain in this case...

**Comment by yeskunall** [comment]
@hferreiro if you have a multi-line comment, and you use `editor::JoinLines` what do you expect a multi-line comment to look like after? Not sure I understand. The editor clearly joined your multi-line comment into one single line? If you’re saying it should use `//` instead of `/*` for denoting the start of a comment, I don’t think that’s a bug.

@ian-h-chamberlain confirmed, yours is still a bug. Thanks for surfacing this!

**Comment by sybrenstuvel** [comment]
This also goes wrong with Python code, using Zed 0.226.5 (`07dae3b7020b0ad79ed91e18c750a16952432866`)

**Edit:** re-tested in Zed 1.0.1+stable.260.ad3e097279b3a5c04b273a4f2f6c6ed1729ea0c8 and it's still an issue.

Joining these lines:
```python
print("hello")
# World
```
Produces this:
```python
print("hello") World
```
which is syntactically incorrect. I expected:
```python
print("hello") # World
```

**Comment by drsybren** [comment]
I've re-checked the [C++](https://github.com/zed-industries/zed/issues/4481#issuecomment-3932535590) and [Python](https://github.com/zed-industries/zed/issues/4481#issuecomment-4038683699) example cases on Zed 1.0.1, and it's still an issue.

**Comment by moechofe** [comment]
I don't know if it's related but joining the line from this c code remove the multiply operator too:
```c
my_var=3
    *4;
```
`:join` produce:
```c
my_var=3 4;
```
